components/operator/api/v1alpha1/serverless_types.go

```diff
@@ -37,7 +46,9 @@ type ServerlessSpec struct {
 	// Used Tracing endpoint
 	Tracing *Endpoint `json:"tracing,omitempty"`
 	// Used Eventing endpoint
-	Eventing       *Endpoint       `json:"eventing,omitempty"`
+	Eventing *Endpoint `json:"eventing,omitempty"`
+
+	// +kubebuilder:validation:XValidation:message="Use dockerRegistry.pv only in combination with dockerRegistry.enableInternal set to true",rule="!has(self.pv) || self.enableInternal"
```

Comment:
It would be great if we had tests for this (like for x-kubernetes validations in `function_types.go`)

Reply:
added